### DISCUSSION TITLE
bkt usage in markdown

### DISCUSSION THREAD

**Discussion by JulienPalard** [dimo414/bkt#16] [Show and tell]
@dimo414 asked me to share how I use bkt in real world, so here it is:

I'm a sometimes teach things (Python mostly) so I'm writing slides from times to times, in Markdown, using [an extremly basic markdown to reveal tool](https://pypi.org/project/mdtoreveal/).

Sometimes I like to write examples, but don't want to spend too much time copy-pasting the commands and their results, so I use a little sed-trick, instead of just giving my file to `mdtoreveal` like:

    mdtoreveal slides.md --output $@

I do use:

    sed 's/#!//e;' slides.md | mdtoreveal /dev/stdin --output $@

So every expression starting with #! will be executed in bash and replaced with their output, for example:
````bash
$ cat example.md 
# Example

```bash
$ python -m timeit -s 'container = set(range(10_000_000))' '10_000_001 in container'
#!python -m timeit -s 'container = set(range(10_000_000))' '10_000_001 in container'
```

$ sed 's/#!//e;' example.md 
# Example

```bash
$ python -m timeit -s 'container = set(range(10_000_000))' '10_000_001 in container'
100 loops, best of 5: 31.5 nsec per loop
```
````

But today I'm writing slides about performance, so I have many commands that take seconds to run, and building my slides soon became very slow: I needed `bkt`!

It's almost as simple as:

```bash
$ python -m timeit -s 'container = set(range(10_000_000))' '10_000_001 in container'
#!bkt -- python -m timeit -s 'container = set(range(10_000_000))' '10_000_001 in container'
```

I just miss a BIG timeout, so I opened a PR to configure bkt from [environment variables](https://github.com/dimo414/bkt/pull/15) to be able to configure it from the `Makefile` instead of having to write `--ttl` `--cache-dir` and so on on each bkt invocation (which I currently do using a little bash wrapper).

**Comment by vorburger** [comment]
> So every expression starting with #! will be executed in bash and replaced with their output, for example:

Cool! just like (my) https://docs.enola.dev/use/execmd/ (for which bkt seems similarly useful)
